packages/mobile/src/escrow/saga.ts

```diff
@@ -174,56 +198,62 @@ function* reclaimFromEscrow(action: ReclaimPaymentAction) {
       yield put(showError(ErrorMessages.RECLAIMING_ESCROWED_PAYMENT_FAILED, ERROR_BANNER_DURATION))
     }
     yield put(reclaimPaymentFailure(e))
-    throw e
   }
 }
 
-function* getEscrowedPayment(paymentID: string) {
-  try {
-    const escrow = yield call(getEscrowContract, web3)
+async function getEscrowedPayment(escrow: Escrow, paymentID: string) {
+  Logger.debug(TAG + '@getEscrowedPayment', 'Fetching escrowed payment')
 
-    Logger.debug(TAG + '@getEscrowedPayment', 'Fetching escrowed payment')
-    const payment = yield escrow.methods.escrowedPayments(paymentID).call()
+  try {
+    const payment = await escrow.methods.escrowedPayments(paymentID).call()
     return payment
   } catch (e) {
     Logger.error(TAG + '@getEscrowedPayment', 'Error fetching escrowed payment', e)
     throw e
   }
 }
 
-function* getSentPayments() {
+function* doFetchSentPayments() {
+  Logger.debug(TAG + '@doFetchSentPayments', 'Fetching valid sent escrowed payments')
+
   try {
-    const escrow = yield call(getEscrowContract, web3)
-    const account = yield call(getConnectedAccount)
-    const recipientsPhoneNumbers = yield select(inviteesSelector)
-    const recipientPhoneNumberToContact = yield select(recipientCacheSelector)
+    const escrow: Escrow = yield call(getEscrowContract, web3)
+    const account: string = yield call(getConnectedAccount)
+    const existingPayments: EscrowedPayment[] = yield select(sentEscrowedPaymentsSelector)
```

Comment:
Are the explicit types necessary?
This questions applies to the other places where explicit types have been added.

Reply:
The explicit types are necessary for the results of `yield` statements. TS doesn't seem to support that atm

Reply:
Ah yes this is sad 😭 
I hope it will change rather sooner.

Reply:
Looks like it's coming soon though! https://github.com/redux-saga/redux-saga/pull/1892